tools/splitblocks/main.go

```diff
@@ -217,17 +225,23 @@ func splitLocalBlock(ctx context.Context, parentDir, blockDir string, meta block
 		level.Info(logger).Log("msg", "splitting block", "minTime", timestamp.Time(minTime), "maxTime", timestamp.Time(maxTime))
 
 		// Inject a modified meta and abuse the repair into removing the now "outside" chunks.
-		// Chunks that cross boundaries are included in multiple blocks.
 		meta.MinTime = minTime
 		meta.MaxTime = maxTime
 		if err := meta.WriteToDir(logger, blockDir); err != nil {
 			return nil, errors.Wrap(err, "failed injecting meta for split")
 		}
-		splitID, err := block.Repair(ctx, logger, parentDir, meta.ULID, block.SplitBlocksSource, block.IgnoreCompleteOutsideChunk, block.IgnoreIssue347OutsideChunk)
+		splitID, err := block.Repair(ctx, logger, parentDir, meta.ULID, block.SplitBlocksSource, true, block.IgnoreCompleteOutsideChunk, block.IgnoreIssue347OutsideChunk)
 		if err != nil {
 			return nil, errors.Wrap(err, "failed while splitting block")
 		}
 
+		if verifyBlocks {
+			blockPath := path.Join(parentDir, splitID.String())
```

Comment:
Nit: This is recalculating splitDir which is right below the if. That could be moved up and reused instead.